src/net.cpp

```diff
@@ -399,6 +401,38 @@ static CAddress GetBindAddress(SOCKET sock)
     return addr_bind;
 }
 
+bool derive_bip324_keys(ECDHSecret&& ecdh_secret, const Span<uint8_t> initiator_hdata, const Span<uint8_t> responder_hdata, BIP324Keys& derived_keys)
+{
+    if (ecdh_secret.size() != ECDH_SECRET_SIZE) {
+        return false;
+    }
+
+    std::string salt{"bitcoin_v2_shared_secret"};
```

Comment:
The BIP currently refers to `BitcoinSharedSecret`. I guess to ack this, i'd have to see a new bip :)

Reply:
:) I am working on re-structuring it. It's close. Will keep you updated here.